src/cmd/sync.go

```diff
@@ -50,23 +50,26 @@ func syncCmd() *cobra.Command {
 	addVerboseFlag, readVerboseFlag := flags.Verbose()
 	addDryRunFlag, readDryRunFlag := flags.DryRun()
 	addAllFlag, readAllFlag := flags.Bool("all", "a", "Sync all local branches", flags.FlagTypeNonPersistent)
+	addNoPushFlag, readNoPushFlag := flags.Bool("no-push", "", "Do not push branches", flags.FlagTypeNonPersistent)
```

Comment:
It wasn't obvious to me from your comment (https://github.com/git-town/git-town/issues/3511#issuecomment-2101035875) whether you wanted `no-push` to be in the `flags` directory for re-use like the `DryRun` flag or if I should add it only to this command. For a smaller diff review, I added it only to this comment.